python/lsst/pipe/tasks/repair.py

```diff
@@ -234,22 +234,40 @@ def cosmicRay(self, exposure, keepCRs=None):
         except Exception:
             pass
 
-        mi = exposure.getMaskedImage()
-        bg = afwMath.makeStatistics(mi, afwMath.MEDIAN).getValue()
+        exposure0 = exposure            # initial value of exposure
+        binSize = self.config.cosmicray.background.binSize
+        nx, ny = exposure.getWidth()/binSize, exposure.getHeight()/binSize
+        if nx*ny <= 1:
+            bg = afwMath.makeStatistics(exposure.getMaskedImage(), afwMath.MEDIAN).getValue()
+            bkgd = None
+        else:
+            exposure = exposure.Factory(exposure, True)
+            bkgd, exposure = measAlg.estimateBackground(exposure, self.config.cosmicray.background,
+                                                      subtract=True)
+            bg = 0.0
 
         if keepCRs is None:
             keepCRs = self.config.cosmicray.keepCRs
         try:
-            crs = measAlg.findCosmicRays(mi, psf, bg, pexConfig.makePolicy(self.config.cosmicray), keepCRs)
+            crs = measAlg.findCosmicRays(exposure.getMaskedImage(),
+                                         psf, bg, pexConfig.makePolicy(self.config.cosmicray), keepCRs)
+            if bkgd:
+                # Add back background image
+                img = exposure.getMaskedImage()
+                img += bkgd.getImageF()
+                del img
+                # Replace original image with CR subtracted image
+                exposure0.getMaskedImage().assign(exposure.getMaskedImage())
```

Comment:
Why not ``exposure0.setMaskedImage(exposure.getMaskedImage())``? That naively seems more obvious. Is there a performance or memory management consideration? I think you end up copying the pixels in both cases, right?